--- .htaccess
+++ .htaccess
@@ -27,6 +27,11 @@ ErrorDocument 404 /index.php
     ErrorDocument 404 /404-error.php
 </FilesMatch>
 
+# This overrides the Drupal 404 handler for files that should never be handled by Drupal
+<FilesMatch "\.(gif|jpe?g|png|bmp|js|css)$">
+    ErrorDocument 404 /404-error.php
+</FilesMatch>
+
 # Set the default handler.
 DirectoryIndex index.php
 
@@ -152,6 +157,12 @@ DirectoryIndex index.php
   RewriteCond %{REQUEST_URI} !^404.%1$
   RewriteRule ^(.*)$ 404.%1 [L]
 
+  RewriteCond %{REQUEST_FILENAME} !-f
+  RewriteCond %{REQUEST_URI} !^.*(sites/.*/imagecache/.*)$
+  RewriteCond %{REQUEST_URI} \.(gif|jpe?g|png|bmp|js|css)$
+  RewriteCond %{REQUEST_URI} !^404.%1$
+  RewriteRule ^(.*)$ 404.%1 [L]
+
   # Rewrite URLs of the form 'x' to the form 'index.php?q=x'.
   RewriteCond %{REQUEST_FILENAME} !-f
   RewriteCond %{REQUEST_FILENAME} !-d
